### ISSUE TITLE
Extract text from candidate packs and add to Applicant object

### ISSUE THREAD

**Issue by tamuri** [UCL-ARC/shortlister#63]
On loading shortlist for first time, extract the text from the candidate pack and 

1. Pull out specific information from the cover sheet, and store appropriate field
    - First Name + Last Name can replace the name currently inferred from the filename
    - Email address, phone number, postcode, region etc. can be stored but don't need to be displayed in details view
    - Answer to question + details for "Do you have the unrestricted right..." should be pulled out and saved in `right_to_work` - True/False
store with in the applicant's Applicant object (`application_text`?)
2. Store the rest of the text of the application (everything but cover page) in Applicant object

Depends on #45

**Comment by shiyingwucl** [comment]
Should I store first name and last name as a separate field?:
```python
@dataclass
class Applicant:
    first_name: str
    last_name: str
```

**Comment by shiyingwucl** [comment]
I can't extract the fields from the candidate pack, but actual body of text seems (slightly) fine:
![Image](https://github.com/user-attachments/assets/7ccc9092-a727-47f6-9bd5-902e15eee00e)

**Comment by tamuri** [comment]
> Should I store first name and last name as a separate field?:

No, I think it's fine to concatenate them

**Comment by tamuri** [comment]
> I can't extract the fields from the candidate pack

Oh, that's interesting. I wonder how the form is rendered.

**Comment by shiyingwucl** [comment]
> > I can't extract the fields from the candidate pack
> 
> Oh, that's interesting. I wonder how the form is rendered.

Update: I think I understand the problem a bit better now. The candidate pack contains an external PIF reference(still not too sure how this works) which is treated as blank by the pdf extraction package. I've tried the following using PyMuPDF:
- Link extraction (not a link)
- Image extraction (not an image either)
- Saving the first pdf as an image (the saved image is correct), and there is a feature in PyMuPDF called "Optical Character Recognition" which could be used to read in text

Maybe I can use the information from page 2 instead?

**Comment by tamuri** [comment]
> candidate pack contains an external PIF reference

Hmmm, interesting. These candidate packs are made up ones, but I wonder whether the real ones would work. Can you share the code to extract the text, please? I will test on a real application (I can't share because of GDPR).

**Comment by shiyingwucl** [comment]
> > candidate pack contains an external PIF reference
> 
> Hmmm, interesting. These candidate packs are made up ones, but I wonder whether the real ones would work. Can you share the code to extract the text, please? I will test on a real application (I can't share because of GDPR).

Sorry! I forgot to update that PyMuPDF's get_text() worked with the candidate pack, I only realised this earlier because there was a huge empty space in the text and jupyter notebook hid the rest of the text that came after. 

I managed to get the text in a more readable format with:
```python
import fitz #pympupdf library

doc = fitz.open("../test_role/Emma_Jones_16743_Candidate_Pack.pdf")  # example document
page = doc[0]
text = page.get_text(sort=True)  # extract sorted words
print(text)
```
![Image](https://github.com/user-attachments/assets/e4d44bb8-654d-440b-8b19-7895e749a3f8)

**Comment by tamuri** [comment]
Ah, that's good. Is fitz an OCR package? I'm worried it will be quite slow.

**Comment by tamuri** [comment]
> Ah, that's good. Is fitz an OCR package?

No, it isn't - tested on a few real candidate packs, and works.

**Comment by shiyingwucl** [comment]
Fitz is the old import name used by pymupdf. I followed a tutorial that used the old name so I just stuck with it 😀

**Comment by tamuri** [comment]
Weird name!
